### PR TITLE
Feature/13280 jo

### PR DESCRIPTION
**Pull request by jeffrey-opdam**
Thank you for your pull request!
By completing this pull request, you agree to the [Contributing License Agreement](https://github.com/ALM-Rangers/Team-Services-Bot/blob/master/.github/CLA.md).

Fixes # .

Changes proposed in this pull request:  
- Added Approvals
- 
- 

@ALM-Rangers/Team-Services-Bot

### CODE REVIEW

File: src/Team-Services-Bot.Api/Dialogs/ApprovalsDialog.cs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+            await Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Replies with a list of ApprovalCards.
+        /// </summary>
+        /// <param name="context">The <see cref="IDialogContext"/>.</param>
+        /// <param name="result">The <see cref="IAwaitable{T}"/>.</param>
+        /// <returns>An async <see cref="Task"/>/.</returns>
+        public async Task ApprovalsAsync(IDialogContext context, IAwaitable<IMessageActivity> result)
```
Comment (jeffrey-opdam) [inline src/Team-Services-Bot.Api/Dialogs/ApprovalsDialog.cs]: Got rid of the wrapper, but needed to make the method public, so the unit tests can easily test them.
Also this way the flow is more clear (with the separate method(s))

File: src/Team-Services-Bot.Api/Dialogs/ApprovalsDialog.cs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            }
+        }
+
+        /// <summary>
+        /// Approves or Rejects an Approval.
+        /// </summary>
+        /// <param name="context">The <see cref="IDialogContext"/>.</param>
+        /// <param name="result">The <see cref="IAwaitable{T}"/>.</param>
+        /// <returns>An async <see cref="Task"/>/.</returns>
+        public async Task ApproveOrReject(IDialogContext context, IAwaitable<IMessageActivity> result)
```
Comment (jeffrey-opdam) [inline src/Team-Services-Bot.Api/Dialogs/ApprovalsDialog.cs]: Same here.

**Comment by jeffrey-opdam** [comment]
I got rid of the wrapper in my unit tests, and better separation of the method(s) for the approval flow.
As soon as this PR is approved. I want to refactor the other tests.

File: src/Team-Services-Bot.Api.UnitTests/Team-Services-Bot.Api.UnitTests.csproj [hunk 3]
```diff
@@ -33,6 +33,7 @@
     <CodeAnalysisRuleSet>..\Common\TestCodeRules.ruleset</CodeAnalysisRuleSet>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <DocumentationFile>bin\Debug\Vsar.TSBot.UnitTests.xml</DocumentationFile>
+    <NoWarn>1591</NoWarn>
```
Comment (Hussar77) [inline src/Team-Services-Bot.Api.UnitTests/Team-Services-Bot.Api.UnitTests.csproj]: Do we really need global suppression? Also, warning should be suppressed for Release configuration as well.

File: src/Team-Services-Bot.Api/Dialogs/ApprovalsDialog.cs (hunk 1 above)
Comment (Hussar77) [inline src/Team-Services-Bot.Api/Dialogs/ApprovalsDialog.cs]: Good simplification

File: src/Team-Services-Bot.Api/Dialogs/ApprovalsDialog.cs (hunk 2 above)
Comment (Hussar77) [inline src/Team-Services-Bot.Api/Dialogs/ApprovalsDialog.cs]: Naming best practice says that asynchronous method name should ends with Async. Please rename to ApproveOrRejectAsync().

**Pull request merged**